test/JoinTest.cpp

```diff
@@ -469,3 +510,256 @@ TEST(JoinTest, invalidJoinVariable) {
 
   ASSERT_ANY_THROW(Join(qec, valuesTree2, valuesTree, 0, 0));
 }
+
+// _____________________________________________________________________________
+TEST(JoinTest, joinTwoLazyOperationsWithAndWithoutUndefValues) {
+  auto performJoin =
+      [](std::vector<IdTable> leftTables, std::vector<IdTable> rightTables,
+         const IdTable& expected, bool expectPossiblyUndefinedResult,
+         ad_utility::source_location loc =
+             ad_utility::source_location::current()) {
+        auto l = generateLocationTrace(loc);
+        auto qec = ad_utility::testing::getQec();
+        RuntimeParameters().set<"lazy-index-scan-max-size-materialization">(0);
+        absl::Cleanup cleanup{[]() {
+          // Reset back to original value to not influence other tests.
+          RuntimeParameters().set<"lazy-index-scan-max-size-materialization">(
+              1'000'000);
```

Comment:
I see. You thought of the issue, but not of the generic solution. The correct way to obtain the original value is to first use `get` before setting to something else.